x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/ProfilingIndexTemplateRegistry.java

```diff
@@ -214,6 +219,22 @@ protected Map<String, ComposableIndexTemplate> getComposableTemplateConfigs() {
         return templatesEnabled ? COMPOSABLE_INDEX_TEMPLATE_CONFIGS : Collections.emptyMap();
     }
 
+    @Override
+    protected boolean isUpgradeRequired(LifecyclePolicy currentPolicy, LifecyclePolicy newPolicy) {
+        try {
+            return getVersion(currentPolicy) < getVersion(newPolicy);
+        } catch (NumberFormatException ex) {
```

Comment:
I've restructured the code now in https://github.com/elastic/elasticsearch/pull/96550/commits/a90bf4314558810caeb4f8dd30f2133f5280cc56. To avoid complicating the control flow I kept the `catch` block here but raise a specific exception now on parse errors in `#getVersion()`.